File: 0392-is-subsequence/0392-is-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
    if (s === "") {
        return true;
    }
    let sPointer = 0;
    let tPointer = 0;
    while (tPointer < t.length) {
        if (t[tPointer] === s[sPointer]) {
            sPointer++;
        }
        if (sPointer === s.length) {
            return true;
        }
        tPointer++;
    }
    return false;
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if s == "":
            return True
        sPointer = 0
        tPointer = 0
        while tPointer < len(t):
            if s[sPointer] == t[tPointer]:
                sPointer += 1
                if sPointer == len(s):
                    return True
            tPointer += 1
        return False